state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2249,9 +2247,7 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		if let Some(tx_id) = deposit_details.deposit_id() {
 			// Only consider rejecting a transaction if we haven't already boosted it,
 			// since by boosting the protocol is committing to accept the deposit.
-			if TransactionsMarkedForRejection::<T, I>::take(broker, &tx_id).is_some() &&
-				!matches!(boost_status, BoostStatus::Boosted { .. })
-			{
+			if TransactionsMarkedForRejection::<T, I>::take(broker, &tx_id).is_some() && !boosted {
```

Comment:
We can avoid reading `TransactionsMarkedForRejection` for boosted transactions by reversing this check (`if !boosted && ..`).

In fact I think we can move the !boosted check into an outer if clause and inline:

```
if !matches!(boost_status, BoostStatus::Boosted { .. }) {
	// check min deposit
	// check for rejection
}
```